src/Optical/LensAssembly.jl

```diff
@@ -324,14 +324,16 @@ Recursive rays are offset by a small amount (`RAY_OFFSET`) to prevent it from im
 
 `trackrays` can be passed an empty vector to accumulate the `LensTrace` objects at each intersection of `ray` with a surface in the assembly.
 """
-function trace(assembly::LensAssembly{T}, r::OpticalRay{T,N}, temperature::T = T(OpticSim.GlassCat.TEMP_REF), pressure::T = T(OpticSim.GlassCat.PRESSURE_REF); trackrays::Union{Nothing,Vector{LensTrace{T,N}}} = nothing, test::Bool = false, recursion::Int = 0)::Union{Nothing,NoPower,LensTrace{T,N}} where {T<:Real,N}
+function trace(system, assembly::LensAssembly{T}, r::OpticalRay{T,N}, temperature::T = T(OpticSim.GlassCat.TEMP_REF), pressure::T = T(OpticSim.GlassCat.PRESSURE_REF); trackrays::Union{Nothing,Vector{LensTrace{T,N}}} = nothing, test::Bool = false, recursion::Int = 0)::Union{Nothing,NoPower,LensTrace{T,N}} where {T<:Real,N}
+    detector_id = surface_id(detector(system))
```

Comment:
do we want to allow for multiple detectors? Maybe detector_id::AbstractVector{T} where{T<:AbstractDetector}.

Reply:
Definitely, haven't done this yet, but multiple detectors are definitely useful.